### ISSUE TITLE
bug: after fixing `pnpmWorkspace` load bug, it resolves wrong folders

### ISSUE THREAD

**Issue by jrson83** [Geocld/sparkee#18]
### Description

Before the recent changes we used [read-yaml-file](https://www.npmjs.com/package/read-yaml-file):

```ts
try {
  const pnpmWorkspace = await readYamlFile<PnpmWorkspace>(fs.readFileSync(PNPM_WORKSPACE, 'utf8'))

  // ... never executed cause `read-yaml-file` expects a `path` not file content & throws error
} catch (err) {
  // always executed
  folders = await glob('packages/*')
}
```

I fixed this changing:

```ts
const pnpmWorkspace = await readYamlFile<PnpmWorkspace>(PNPM_WORKSPACE)
let wPackages = pnpmWorkspace.packages

// this code now is a new problem
wPackages = wPackages.map((wp) => {
  return `${wp.split('/')[0]}/*`
})

await Promise.all(
  wPackages.map(async (wp) => {
    const wFolders = await glob(wp)
      folders = folders.concat(wFolders)
  })
)
```

Now the problem is that [pnpm-workspace.yaml](https://pnpm.io/pnpm-workspace_yaml) can have multiple entries, with and without globs `*` and with and without excludes `!`.

```yml
packages:
  - 'docs'
  - 'another-single-package'
  - 'packages/*'
  - 'multiple-sub-packages/**/*'
  - '!**/__tests__/**'
```

After reading the yml, `pnpmWorkspace` is:

```ts
{
  packages: [
    'docs',
    'another-single-package',
    'packages/*',
    'multiple-sub-packages/**/*',
    '!**/__tests__/**'
  ]
}
```

After the map function `folders` is:

```ts
[
  'docs/node_modules',
  'docs/package.json',
  'docs/src',
  'docs/tsconfig.json',
  'packages/one',
  'packages/two',
  'multiple-sub-packages/another-subfolder',
  'multiple-sub-packages/subfolder'
]
```

Now the glob function fails and throws:

```sh
[Error: ENOENT: no such file or directory, open 'docs/node_modules/package.json'] {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'open',
  path: '/docs/node_modules/package.json'
}
```

**Comment by Geocld** [comment]
@jrson83 thanks, I got it. To fix this bug, Is it possible to detect there is a package.json file before we use glob?

**Comment by jrson83** [comment]
@Geocld I think the fix is to remove the map function and instead use a parameter `onlyDirectories` if exists for the glob function.

**Comment by Geocld** [comment]
@jrson83 okey, i will try fix it in next version! you can also fix it If you have more smart idea :)

**Comment by jrson83** [comment]
@Geocld it is all totally broken at the moment and ruined my changelog and npm release.

**Comment by Geocld** [comment]
@jrson83 sorry I am too busy recently, this issue had fixed in [1.4.0](https://www.npmjs.com/package/sparkee/v/1.4.0), If you still have any questions, please feel free to leave me a message.
